test/test_buffer.py

```diff
@@ -176,6 +176,6 @@ def test_contains():
 @with_setup(setup=cleanup)
 def test_set_items_for_range():
     vim.current.buffer[:] = ['a', 'b', 'c', 'd', 'e']
-    r = vim.current.buffer.range(1, 3)
+    r = vim.current.buffer.range(0, 3)
```

Comment:
Won't this break existing plugins that use python-client?

Reply:
yes, this would be an API-break to make range generation and buffer-indexing and such consistent overall with python-indexing starting at 0 (and make the buffer[a, b] and buffer.range(a, b) consistent as well in the process.

It is definitely an API-break, that's why I prepended "Proposal" to it because I'd like to know upstreams opinion on this matter.

I think that API-consistency is worth the API-break, but it's not on me to decide that as I am not the maintainer. :)

Reply:
It's not really inconsistent, Range is a Vim concept, see `:help :range`. Buffers start with line 1.

`buffer[a:b]` is treated differently because it's considered a "list" not a "buffer", just like if you had a list of lines in VimL it would be zero-indexed.

Reply:
Ah, ok, then the current API makes perfect sense. Thanks for clarifying!